geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -482,5 +481,43 @@ class MapListFunctionFactory extends TransformerFunctionFactory {
         }.toMap.asJava
     }
   }
-
+}
+class CastFunctionFactory extends TransformerFunctionFactory {
+  override def functions = Seq(string2double,string2int,string2float,string2long,string2boolean)
+
+  val string2double = TransformerFn("string2double") {
```

Comment:
@drackaer will the empty/null handling apply to converters with casting? I think that's what we originally wanted to do here instead of explicitly calling string2double when i have something like:

```
 converter = {
   type         = "delimited-text",
   format       = "DEFAULT",
   id-field     = "md5(string2bytes($0))",
   fields = [
     { name = "oneup",    transform = "$1" },
     { name = "phrase",   transform = "$2" },
     { name = "somedub",   transform = "$3::double"     },
     { name = "geom",     transform = "geometry($4)"   }
   ]
 }
```

data:
```
1,hello,3.0,Point(46.0 45.0)
2,world,,Point(90.0 90.0)
```

If it handles that can you add explicit tests for that?